### PR TITLE
SQL query is printed into browser in case of exception

### PR DESCRIPTION
**Pull request by shyamranpara**
### Description
SQL query is printed into browser in case of exception

### Fixed Issues (if relevant)

1. magento/magento2#13385: SQL query is printed into browser in case of exception

### Manual testing scenarios
Add an erroneous SQL statement to collection select object. I've added this code to \Magento\Catalog\Block\Product\ListProduct::initializeProductCollection():
$collection = $layer->getProductCollection();
$collection->getSelect()->columns('qwerty');
as an example
Open some category page in a browser.
SQL query is printed in a browser

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by shyamranpara** [comment]
@magento-engcom-team, can you guide me how to remove old commits not related to this ?
Custom address attribute shown on frontend #12524
Travis cli fix

**Comment by hostep** [comment]
@shyamranpara: every PR needs its own branch and you should try to make sure it is up2date with the current `2.2-develop` branch from the magento repo. Then create a new commit on that new branch and then create a PR from that new branch.
To make sure your new branch is up2date with the latest from the magento repo, is to add an `upstream` remote and bring your local branch in sync with the `upstream` repo, you can read more about this over here:
- https://help.github.com/articles/configuring-a-remote-for-a-fork/
- https://help.github.com/articles/syncing-a-fork/ (replace `upstream/master` with `upstream/2.2-develop`

There is also some documentation on the devdocs, but they lack actual examples:
http://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#sync

I tend to do something like this after the `upstream` remote has been set up:
```bash
git checkout 2.2-develop # checkout the 2.2-develop branch on my local machine
git fetch upstream # fetch all the changes from the upstream remote (they aren't being applied just yet)
git pull upstream 2.2-develop # bring my local 2.2-develop branch up2date with the one from the magento repo
git checkout -b my-new-branch # create a new branch with the name 'my-new-branch' based on the current branch in use
# manipulate some files
git commit -m "My new commit which fixes A & B"
git push origin my-new-branch # push your new branch to your own fork (=origin remote)
# now create a PR using Github's UI
```

Hope this helps

**Comment by orlangur** [comment]
@shyamranpara even simpler approach is to create a new branch based on `2.2-develop` as shown in previous comment and then do `git cherry-pick d77b9708dcc361c3f849219b24b41ba11b564384`.

**Comment by shyamranpara** [comment]
@orlangur , @hostep  Thanks for guide.
I created new PR https://github.com/magento/magento2/pull/13623 hence, I guess this PR should be closed or maked as rejected.

**Comment by orlangur** [comment]
@shyamranpara new PR contains two commits anyway, let's continue here. Please just force push into the same `issue-13385` branch.

**Comment by orlangur** [comment]
@shyamranpara please rewrite branch so that it contains only necessary changes as a single commit.

Just tell me if you need any assistance with that.

**Comment by shyamranpara** [comment]
thanks for support let me try again.

**Comment by shyamranpara** [comment]
@orlangur I have rewrite branch please check it.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-839](https://jira.corp.magento.com/browse/ENGCOM-839) has been created to process this Pull Request
